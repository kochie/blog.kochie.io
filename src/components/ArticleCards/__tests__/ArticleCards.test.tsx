import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'
import {
  CardDetails,
  LargeCard,
  MediumCard,
  SmallCard,
} from '@/components/ArticleCards'

const cardDetails: CardDetails = {
  title: 'title',
  image: {
    lqip: 'AAAAAAAAAAAA',
    url: 'https://blog.kochie.io/test.webp',
    alt: 'alt',
  },
  blurb: 'the blurb',
  readTime: '1 min read',
  tags: ['tag1'],
  articleDir: 'articleDir',
}

describe('ARTICLECARDS LARGE COMPONENT', () => {
  test('renderes correctly', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(<LargeCard {...cardDetails} />)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

describe('ARTICLECARDS MEDIUM COMPONENT', () => {
  test('renderes correctly', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(<MediumCard {...cardDetails} />)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

describe('ARTICLECARDS SMALL COMPONENT', () => {
  test('renderes correctly', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(<SmallCard {...cardDetails} />)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
