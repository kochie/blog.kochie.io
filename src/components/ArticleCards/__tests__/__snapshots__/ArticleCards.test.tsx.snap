// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ARTICLECARDS LARGE COMPONENT renderes correctly 1`] = `
<div
  className="md:flex w-full h-full col-span-6 group shadow-sm hover:shadow-2xl dark:shadow-none transition ease-in-out duration-200 rounded-2xl"
>
  <div
    className="shadow-2xl rounded-2xl transition ease-in-out duration-200 bg-gray-300 dark:bg-neutral-600 text-black dark:text-white w-full h-full z-20 dark:shadow-neutral-800"
  >
    <div
      className="flex flex-col md:flex-row"
    >
      <div
        className="relative rounded-t-2xl md:rounded-l-2xl md:rounded-tr-none md:w-1/2 lg:w-2/3 overflow-hidden"
      >
        <a
          href="/articles/articleDir"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="transition ease-in-out duration-200 group-hover:grayscale-0 grayscale-30 filter"
          >
            <img
              alt="alt"
              className="rounded-t-2xl w-full h-52 md:rounded-l-2xl md:rounded-tr-none md:h-80 group-hover:scale-110 transform-gpu transition ease-in-out duration-200 cursor-pointer"
              data-nimg="1"
              decoding="async"
              height={300}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
              src="/_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=1200&q=75"
              srcSet="/_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=320&q=75 320w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=420&q=75 420w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=768&q=75 768w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=1024&q=75 1024w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=1200&q=75 1200w"
              style={
                {
                  "backgroundImage": "url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http%3A//www.w3.org/2000/svg' viewBox='0 0 600 300'%3E%3Cfilter id='b' color-interpolation-filters='sRGB'%3E%3CfeGaussianBlur stdDeviation='20'/%3E%3C/filter%3E%3Cimage preserveAspectRatio='none' filter='url(%23b)' x='0' y='0' height='100%25' width='100%25' href='AAAAAAAAAAAA'/%3E%3C/svg%3E")",
                  "backgroundPosition": "50% 50%",
                  "backgroundRepeat": "no-repeat",
                  "backgroundSize": "cover",
                  "color": "transparent",
                  "height": "400px",
                  "objectFit": "cover",
                  "width": "100%",
                }
              }
              width={600}
            />
          </div>
        </a>
        <div
          className="absolute p-4 bottom-0 md:right-0 md:top-0 md:p-4 md:bottom-0"
          style={
            {
              "height": "min-content",
            }
          }
        >
          <div
            className="undefined tagSet"
          >
            <a
              href="/tags/tag1"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="tag tag-common"
              >
                tag1
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="p-4 lg:w-1/3 relative"
      >
        <a
          href="/articles/articleDir"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <h2
            className="heading text-2xl"
          >
            title
          </h2>
        </a>
        <p
          className="mt-2 mb-10"
        >
          the blurb
        </p>
        <div
          className="text-right absolute bottom-0 right-0 p-6"
        >
          <sub>
            1 min read
          </sub>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ARTICLECARDS MEDIUM COMPONENT renderes correctly 1`] = `
<div
  className="md:col-span-3 col-span-6 w-full h-full group"
>
  <div
    className="shadow-2xl rounded-2xl transition ease-in-out duration-200 bg-gray-300 dark:bg-neutral-600 text-black dark:text-white w-full h-full z-20 dark:shadow-neutral-800"
  >
    <div
      className="flex flex-col relative h-full"
    >
      <div
        className="bg-black rounded-t-2xl relative overflow-hidden"
      >
        <div
          className="transition ease-in-out duration-200 group-hover:grayscale-0 grayscale-30 filter"
        >
          <img
            alt="alt"
            className="rounded-t-2xl group-hover:scale-110 transform-gpu transition ease-in-out duration-200 cursor-pointer"
            data-nimg="1"
            decoding="async"
            height={300}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            sizes="100vw"
            src="/_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=1200&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=320&q=75 320w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=420&q=75 420w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=768&q=75 768w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=1024&q=75 1024w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=1200&q=75 1200w"
            style={
              {
                "backgroundImage": "url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http%3A//www.w3.org/2000/svg' viewBox='0 0 600 300'%3E%3Cfilter id='b' color-interpolation-filters='sRGB'%3E%3CfeGaussianBlur stdDeviation='20'/%3E%3C/filter%3E%3Cimage preserveAspectRatio='none' filter='url(%23b)' x='0' y='0' height='100%25' width='100%25' href='AAAAAAAAAAAA'/%3E%3C/svg%3E")",
                "backgroundPosition": "50% 50%",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "cover",
                "color": "transparent",
                "height": "300px",
                "objectFit": "cover",
                "width": "100%",
              }
            }
            width={600}
          />
        </div>
        <div
          className="absolute p-4 bottom-0"
        >
          <div
            className="undefined tagSet"
          >
            <a
              href="/tags/tag1"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="tag tag-common"
              >
                tag1
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="p-4"
      >
        <a
          href="/articles/articleDir"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <h2
            className="heading text-2xl"
          >
            title
          </h2>
        </a>
        <p
          className="mb-8 mt-2"
        >
          the blurb
        </p>
        <div
          className="text-right absolute bottom-0 right-0 p-4"
        >
          <sub>
            1 min read
          </sub>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ARTICLECARDS SMALL COMPONENT renderes correctly 1`] = `
<div
  className="md:col-span-3 lg:col-span-2 w-full h-full col-span-6 group"
>
  <div
    className="shadow-2xl rounded-2xl transition ease-in-out duration-200 bg-gray-300 dark:bg-neutral-600 text-black dark:text-white w-full h-full z-20 dark:shadow-neutral-800"
  >
    <div
      className="flex flex-col h-full"
    >
      <div
        className="relative bg-black rounded-t-2xl overflow-hidden"
      >
        <div
          className="transition ease-in-out duration-200 group-hover:grayscale-0 grayscale-30 filter"
        >
          <img
            alt="alt"
            className="rounded-t-2xl group-hover:scale-110 transform-gpu transition ease-in-out duration-200 cursor-pointer"
            data-nimg="1"
            decoding="async"
            height={300}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
            src="/_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=1200&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=320&q=75 320w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=420&q=75 420w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=768&q=75 768w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=1024&q=75 1024w, /_next/image?url=https%3A%2F%2Fblog.kochie.io%2Ftest.webp&w=1200&q=75 1200w"
            style={
              {
                "backgroundImage": "url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http%3A//www.w3.org/2000/svg' viewBox='0 0 600 300'%3E%3Cfilter id='b' color-interpolation-filters='sRGB'%3E%3CfeGaussianBlur stdDeviation='20'/%3E%3C/filter%3E%3Cimage preserveAspectRatio='none' filter='url(%23b)' x='0' y='0' height='100%25' width='100%25' href='AAAAAAAAAAAA'/%3E%3C/svg%3E")",
                "backgroundPosition": "50% 50%",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "cover",
                "color": "transparent",
                "height": "300px",
                "objectFit": "cover",
                "width": "100%",
              }
            }
            width={600}
          />
        </div>
        <div
          className="absolute p-4 bottom-0"
        >
          <div
            className="undefined tagSet"
          >
            <a
              href="/tags/tag1"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              <div
                className="tag tag-common"
              >
                tag1
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        className="p-4 relative flex-grow"
      >
        <a
          href="/articles/articleDir"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <h2
            className="heading text-2xl"
          >
            title
          </h2>
        </a>
        <p
          className="mt-2 mb-10"
        >
          the blurb
        </p>
        <div
          className="text-right absolute bottom-0 right-0 p-6"
        >
          <sub>
            1 min read
          </sub>
        </div>
      </div>
    </div>
  </div>
</div>
`;
